src/plugins/data/common/index_patterns/types.ts

```diff
@@ -17,19 +17,68 @@
  * under the License.
  */
 
-import { IFieldType } from './fields';
+import { IFieldList, Field, IFieldType } from '../../public';
 
 export interface IIndexPattern {
```

Comment:
Sorry if I didn't make this clear beforehand
It wasn't the intention of `IIndexPattern` to be a full representation of all fields and methods on `IndexPattern`.
It's just a static representation without some of the methods like `create` and `destroy`.
If you require access to those, you should still use `IndexPattern` directly.

Reply:
d143c22